handlers/dart/dart.go

```diff
@@ -184,16 +185,16 @@ func (h *handler) Send(ctx context.Context, msg courier.Msg, clog *courier.Chann
 			return status, nil
 		}
 
-		responseText := string(respBody)
-		if responseText != "000" {
-			errorMessage := "Unknown error"
-			if responseText == "001" {
-				errorMessage = "Error 001: Authentication Error"
+		responseCode := stringsx.Truncate(string(respBody), 3)
+		if responseCode != "000" {
+			errorMessage := "Unknown error."
+			if responseCode == "001" {
+				errorMessage = "Authentication error."
+			} else if responseCode == "101" {
+				errorMessage = "Account expired or invalid parameters."
 			}
-			if responseText == "101" {
-				errorMessage = "Error 101: Account expired or invalid parameters"
-			}
-			clog.Error(fmt.Errorf(errorMessage))
+
+			clog.Error(courier.ErrorServiceSpecific("dart", responseCode, errorMessage))
```

Comment:
Here we know what some of the error codes mean so we can generate some error messages.. but we want to avoid having to interpret all the possible error codes we can get back from a provider like Twilio, so we can just pass an empty message to use a generic one, and then point users to the external site.